src/sys/sendfile.rs

```diff
@@ -125,6 +120,45 @@ cfg_if! {
                 )
             }
         }
+    } else if #[cfg(solarish)] {
+        use std::os::unix::io::BorrowedFd;
+        use std::marker::PhantomData;
+
+        #[derive(Debug, Copy, Clone)]
+        /// Mapping of the raw C sendfilevec_t struct
+        pub struct SendfileVec<'a> {
+            raw: libc::sendfilevec_t,
+            phantom: PhantomData<BorrowedFd<'a>>
+        }
+
+        const SFV_FD_SELF: i32 = -2;
+
+        impl<'a> SendfileVec<'a> {
```

Comment:
> We should probably use BorrowedFd<'a> here so that it is consistent with FdSet and PollFd

BTW, I think we should also do this to Epoll

You re seing an old version.

Reply:
Emmm, why did you revert the definition of `SendfileVec`, for that `BorrowedFd`, I was just suggesting changing the function parameter type to `BorrowedFd<'fd>`

Reply:
Just like what we did to `FdSet`: https://github.com/nix-rust/nix/blob/39ad47bc80af68fb9ad67d37ab699684bea0e2a9/src/sys/select.rs#L18-L47